src/app/application.cpp

```diff
@@ -222,6 +224,24 @@ MainWindow *Application::mainWindow()
     return m_window;
 }
 
+WindowState Application::initialWindowState() const
+{
```

Comment:
Why all this convoluted stuff instead of using the loaded value in `m_initialWindowState`?
Isn't the purpose of this PR to decouple the minimize settings from the startup behavior?
I think the code below belongs to upgrade.cpp

Reply:
> I think the code below belongs to upgrade.cpp

OK.
All suggested changes are done.